src/HSE/Util.hs

```diff
@@ -393,28 +394,18 @@ extensionImpliedBy :: Extension -> [Extension]
 extensionImpliedBy = \x -> Map.findWithDefault [] x mp
     where mp = Map.fromListWith (++) [(b, [a]) | (a,bs) <- extensionImplications, b <- bs]
 
--- | (a, bs) means extension a implies all of bs.
---   Taken from https://downloads.haskell.org/~ghc/master/users-guide/glasgow_exts.html#language-options
---   In the GHC source at DynFlags.impliedXFlags
+-- | (a, bs) means extension a implies all of bs. Uses GHC source at
+-- DynFlags.impliedXFlags
 extensionImplications :: [(Extension, [Extension])]
-extensionImplications = map (first EnableExtension) $
-    (RebindableSyntax, [DisableExtension ImplicitPrelude]) :
-    map (second (map EnableExtension))
-    [ (DerivingVia              , [DerivingStrategies])
-    , (RecordWildCards          , [DisambiguateRecordFields])
-    , (ExistentialQuantification, [ExplicitForAll])
-    , (FlexibleInstances        , [TypeSynonymInstances])
-    , (FunctionalDependencies   , [MultiParamTypeClasses])
-    , (GADTs                    , [MonoLocalBinds])
-    , (IncoherentInstances      , [OverlappingInstances])
---    Incorrect, see https://github.com/ndmitchell/hlint/issues/587
---    , (ImplicitParams           , [FlexibleContexts, FlexibleInstances])
-    , (ImpredicativeTypes       , [ExplicitForAll, RankNTypes])
-    , (LiberalTypeSynonyms      , [ExplicitForAll])
-    , (PolyKinds                , [KindSignatures])
-    , (RankNTypes               , [ExplicitForAll])
-    , (ScopedTypeVariables      , [ExplicitForAll])
-    , (TypeOperators            , [ExplicitNamespaces])
-    , (TypeFamilies             , [ExplicitNamespaces, KindSignatures, MonoLocalBinds])
-    , (TypeFamilyDependencies   , [ExplicitNamespaces, KindSignatures, MonoLocalBinds, TypeFamilies])
-    ]
+extensionImplications = map (toHse . g) (foldl' f [] impliedXFlags)
+  where
+    f m (e, _, _) = if e `notElem` m then e : m else m
```

Comment:
Does this end up being O(n^2)? Would it be better to just do `e : m` and then a `nubOrd` at the end?

Reply:
Can't `nubOrd` (no `Ord`). I realize now that I've missed a spot though where we should `nub`.

Reply:
That concerns me a lot - these extensions could get numerous. An O(n^2) here could end up being noticable. Any chance we send an upstream patch to GHC to derive Ord on these things?

Reply:
I don't see why that should be hard. I'll take it.

Reply:
MR : https://gitlab.haskell.org/ghc/ghc/merge_requests/2707.